components/solr-access-service/api/src/main/java/org/phenotips/solr/HPOScriptService.java

```diff
@@ -78,12 +78,16 @@
             crt = nodes.poll();
             results.add(String.valueOf(crt.get(ID_FIELD_NAME)));
             @SuppressWarnings("unchecked")
-            List<String> parents = (List<String>) crt.get("is_a");
+            Object parents = crt.get("is_a");
             if (parents == null) {
                 continue;
             }
-            for (String pid : parents) {
-                nodes.add(this.get(StringUtils.substringBefore(pid, " ")));
+            if (parents.getClass() == String.class) {
```

Comment:
use [`instanceof`](http://docs.oracle.com/javase/tutorial/java/nutsandbolts/op2.html) operator instead of class comparison